backend/src/util/VerifyFIN.ts

```diff
@@ -0,0 +1,34 @@
+const checkAlpha = ['X', 'W', 'U', 'T', 'R', 'Q', 'P', 'N', 'M', 'L', 'K'];
+const weightArray = [2, 7, 6, 5, 4, 3, 2];
+
+function validateFIN(fin: string): string {
+  const finUpper = fin.trim.toUpperCase();
+  if (finUpper.length === 9) {
```

Comment:
Reduce the nesting by returning false if the length is not 9.

Reply:
Sorry, could you elaborate? don't quite what get this means :(

Reply:
This is pattern called [early return](https://medium.com/swlh/return-early-pattern-3d18a41bba8). The idea is that code becomes more difficult to understand every time you add another level of `if` or `for` etc (this is called nesting).

We'll look at this function as an example from L19-24, we have these lines:

```js
      return true;
    }
    return false;
  }
  return false;
}
```

At one glance, I can't tell what is what. Does the first return belong to `if (finUpper.length === 9) {`? or is it the second one? And I also have to read past the entire body of the `if` only to realise at the end that I just return false.

To apply this pattern, think of your code as 'exiting' early if the conditions are not met. You want to do all of the heavy lifting at the end of the function instead of making your reviewer read everything up front and then worrying whether or not there is more afterwards.

In this case:

```js
function validateFin(fin) {
  if (fin.length !== 9) {
    return false;
  }
  //..... check the checksum
  return isChecksumValid;
}
```

Notice how we no longer have `if` in `if` and I can immediately find out where the important stuff is.